--- a/base/third_party/symbolize/symbolize.cc
+++ b/base/third_party/symbolize/symbolize.cc
@@ -57,6 +57,7 @@
 #if defined(HAVE_SYMBOLIZE)
 
 #include <string.h>
+#include <utility>
 
 #include <algorithm>
 #include <limits>
--- a/base/third_party/symbolize/symbolize.h
+++ b/base/third_party/symbolize/symbolize.h
@@ -61,6 +61,7 @@
 #ifdef HAVE_SYMBOLIZE
 
 #include <algorithm>
+#include <utility>
 
 #if defined(__ELF__)  // defined by gcc
 #if defined(__OpenBSD__)
--- a/v8/src/heap/cppgc/prefinalizer-handler.h
+++ b/v8/src/heap/cppgc/prefinalizer-handler.h
@@ -6,6 +6,7 @@
 #define V8_HEAP_CPPGC_PREFINALIZER_HANDLER_H_
 
 #include <vector>
+#include <utility>
 
 #include "include/cppgc/prefinalizer.h"
 
