--- a/base/third_party/symbolize/symbolize.cc
+++ b/base/third_party/symbolize/symbolize.cc
@@ -57,6 +57,7 @@
 #if defined(HAVE_SYMBOLIZE)
 
 #include <string.h>
+#include <utility>
 
 #include <algorithm>
 #include <limits>
--- a/base/third_party/symbolize/symbolize.h
+++ b/base/third_party/symbolize/symbolize.h
@@ -61,6 +61,7 @@
 #ifdef HAVE_SYMBOLIZE
 
 #include <algorithm>
+#include <utility>
 
 #if defined(__ELF__)  // defined by gcc
 #if defined(__OpenBSD__)
--- a/net/base/address_list.h
+++ b/net/base/address_list.h
@@ -10,6 +10,7 @@
 #include <string>
 #include <utility>
 #include <vector>
+#include <tuple>
 
 #include "base/compiler_specific.h"
 #include "net/base/ip_endpoint.h"
--- a/media/capabilities/webrtc_video_stats_db.h
+++ b/media/capabilities/webrtc_video_stats_db.h
@@ -7,6 +7,7 @@
 
 #include <memory>
 #include <string>
+#include <vector>
 
 #include "base/callback_forward.h"
 #include "base/check.h"
