--- a/build/config/compiler/BUILD.gn.orig
+++ b/build/config/compiler/BUILD.gn
@@ -1631,10 +1631,6 @@ config("default_warnings") {
           # TODO(https://crbug.com/1028110): Evaluate and possible enable.
           "-Wno-deprecated-copy",
 
-          # TODO(https://crbug.com/1203071): Clean up and enable.
-          "-Wno-unused-but-set-parameter",
-          "-Wno-unused-but-set-variable",
-
           # TODO(https://crbug.com/1239077): Fix and re-enable.
           "-Wno-inline-asm",
         ]
