--- a/build/config/compiler/BUILD.gn.orig
+++ b/build/config/compiler/BUILD.gn
@@ -1523,18 +1523,6 @@
           "-Wno-builtin-assume-aligned-alignment",
         ]
 
-        # NaCl does not support flags from ToT.
-        if (!is_nacl) {
-          cflags += [
-            # TODO(https://crbug.com/1203071): Clean up and enable.
-            "-Wno-unused-but-set-parameter",
-            "-Wno-unused-but-set-variable",
-
-            # TODO(https://crbug.com/1255745): Clean up, enable.
-            "-Wno-bitwise-instead-of-logical",
-          ]
-        }
-
         if (is_fuchsia) {
           # TODO(https://bugs.chromium.org/p/fuchsia/issues/detail?id=77383)
           cflags += [ "-Wno-deprecated-copy" ]
